gtime/forecasting/tests/test_simple_models.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+import pandas as pd
+import pytest
+from pandas.util import testing as testing
+from hypothesis import given, note
+import hypothesis.strategies as st
+from gtime.utils.hypothesis.time_indexes import giotto_time_series
+from gtime.model_selection import horizon_shift, FeatureSplitter
+
+from gtime.forecasting import (
+    NaiveModel,
+    SeasonalNaiveModel,
+    DriftModel,
+    AverageModel
+)
+
+
+@st.composite
+def forecast_input(draw, max_lenth):
+    length = draw(st.integers(min_value=2, max_value=max_lenth))
+    horizon = draw(st.integers(min_value=1, max_value=length - 1))
+    X = draw(giotto_time_series(
+        min_length=length,
+        max_length=max_lenth,
+        allow_nan=False,
+        allow_infinity=False
+    ))
+    y = horizon_shift(X, horizon=horizon)
+    X_train, y_train, X_test, y_test = FeatureSplitter().transform(X, y)
+    return X_train, y_train, X_test
+
+
+class SimplePipelineTest:
+
+    def setup(self, data, Model):
```

Comment:
I don't understand well this structure with the `setup` method. In general `pytest` encourages the usage of `pytest.fixture`.  Is it possible to refactor it like that or is there a specific reason why you used setup?

Reply:
The idea of setup itself was to run fit/predict once, then save the results (model, y_pred) and run tests on them without recalculating. It can probably be done with fixtures too. One of the points of such structure is it's easy to set up new classes inheriting the functions from the first one and only adding model parameters, like:
```
    @given(data=forecast_input(50), season_length=st.integers(min_value=1, max_value=100))
    def setup(self, data, season_length):
        self.season_length = season_length
        super().setup(data, SeasonalNaiveForecaster(seasonal_length=season_length))
```
Do you think the same can be done with `fixtures`? I'm not much of an expert with them

Reply:
Ah I see, so if you use `hypothesis` in the setup of the child class I think it is not compatible with fixtures.
Otherwise, otherwise I think it is better to stick with fixtures.